src/wallets/sendTransaction/SendAdvancedTransaction.tsx

```diff
@@ -16,33 +17,64 @@ import {
   toAntValidator,
   validateAddressOrEmpty,
 } from '../../common/util'
+import {PropsWithWalletState, withStatusGuard} from '../../common/wallet-status-guard'
+import {getSendTransactionParams} from './ConfirmAdvancedTransaction'
+import {asEther, asWei, Wei} from '../../common/units'
+import {rendererLog} from '../../common/logger'
 
 interface SendAdvancedTransactionProps {
   onSend: () => void
+  availableAmount: Wei
   transactionParams: AdvancedTransactionParams
   setTransactionParams: (advancedParams: Partial<AdvancedTransactionParams>) => void
   estimateTransactionFee: () => Promise<FeeEstimates>
   onCancel: () => void
 }
 
-export const SendAdvancedTransaction: FunctionComponent<SendAdvancedTransactionProps &
-  ModalProps> = ({
+const _SendAdvancedTransaction = ({
+  availableAmount,
   transactionParams,
   setTransactionParams,
   onCancel,
   onSend,
-}: SendAdvancedTransactionProps & ModalProps) => {
+  walletState,
+}: PropsWithWalletState<SendAdvancedTransactionProps & ModalProps, LoadedState>): JSX.Element => {
   const {t} = useTranslation()
   const modalLocker = ModalLocker.useContainer()
+  const {estimateGas} = walletState
 
   const {amount, recipient, gasLimit, gasPrice, data, nonce} = transactionParams
 
+  const [gasEstimate, setGasEstimate] = useState<undefined | number>(undefined)
+
+  useEffect(() => {
+    async function doGasEstimate(): Promise<void> {
+      try {
+        const estimatedGas = await estimateGas(getSendTransactionParams(transactionParams))
+        setGasEstimate(estimatedGas)
+      } catch (e) {
+        rendererLog.error(e)
+        setGasEstimate(undefined)
+      }
+    }
+    doGasEstimate()
+  }, [transactionParams])
+
   const addressValidator = toAntValidator(t, validateAddressOrEmpty)
   const txAmountValidator = createAdvancedTxAmountValidator(t)
   const gasAmountValidator = createGasAmountValidator(t)
   const feeValidator = createFeeValidator(t)
   const hexValidator = createHexValidator(t)
 
+  const feeLimit = new BigNumber(gasLimit).times(gasPrice)
+  const totalAmount = asEther(new BigNumber(amount).plus(feeLimit))
+
+  const remainingBalance = asWei(
+    totalAmount.isFinite() ? availableAmount.minus(totalAmount) : availableAmount,
+  )
+
+  const disableSend = !remainingBalance.isPositive()
```

Comment:
`!x.isZero() && x.isNegative()` ?

Reply:
Isn't it the same as simply `x.isNegative()`?

Reply:
It's not the same for `-0` (yes). I'm not sure though if it's possible to get that value here.
It should be ok to simply use `isNegative`.

Reply:
I see, we can do `x.isLessThan(0)` then which covers the case with negative zero.